Tests broken for Python 2.7 and 3.5

#### Summary
Several tests are suddenly failing for Python 2.7 and Python 3.5.
https://travis-ci.org/release-engineering/ubi-population-tool/builds/588606914?utm_source=github_status&utm_medium=notification


#### Issue Type
- Bug Report

#### Steps to reproduce
Run CI suite by creating a pull request.
Tested here: #133 

#### Actual results
Tests all seem to fail with the following;
E           ValueError: No JSON object could be decoded
/opt/python/2.7.15/lib/python2.7/json/decoder.py:382: ValueError
------------------------------ Captured log call -------------------------------
WARNING  urllib3.connectionpool:connectionpool.py:298 Connection pool is full, discarding connection: foo.pulp.com

#### Expected results
Tests should all pass successfully.

#### OS / Environment
Travis CI

#### Additional Information
Failing tests:
tests/test_pulp.py::test_retries[True-500-None-search_repo_by_cs-retry_args0-{}-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-search_repo_by_id-retry_args1-{}-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-search_rpms-retry_args2-[]-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-search_modules-retry_args3-[]-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-wait_for_tasks-retry_args4-{"state":"finished","task_id":"fake-tid"}-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-search_tasks-retry_args5-[]-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-unassociate_units-retry_args6-{"spawned_tasks":[]}-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-associate_units-retry_args7-{"spawned_tasks":[]}-10] FAILED
tests/test_pulp.py::test_retries[True-500-None-publish_repo-retry_args8-{"spawned_tasks":[]}-10] FAILED
tests/test_pulp.py::test_retries[True-500-3-search_repo_by_cs-retry_args9-{}-3] FAILED

--- a/tests/test_pulp.py
+++ b/tests/test_pulp.py
@@ -2,13 +2,14 @@ import sys
 import threading
 
 try:
-    from http.client import HTTPMessage
+    from http.client import HTTPResponse
 except ImportError:
-    from httplib import HTTPMessage
+    from httplib import HTTPResponse
+
+from io import BytesIO
 
 import pytest
 import requests
-import six
 
 from mock import MagicMock, patch
 from requests.exceptions import HTTPError
@@ -298,12 +299,21 @@ def fixture_mocked_getresponse():
         yield mocked_get_conn.return_value.getresponse
 
 
-def make_mock_response(status, text=None):
-    m = MagicMock(name='MockResponse', status=status)
-    m.isclosed.side_effect = [False, True]
-    m.msg = MagicMock(spec=HTTPMessage, headers=[], status=status)
-    m.read.return_value = six.b(text)
-    return m
+def make_mock_response(status, text):
+    response_string = (
+        "HTTP/1.1 {0} Reason\r\n"
+        "Content-Type: application/json\r\n"
+        "\r\n"
+        "{1}"
+    ).format(status, text)
+
+    mocked_sock = MagicMock()
+    mocked_sock.makefile.return_value = BytesIO(response_string.encode())
+
+    http_response = HTTPResponse(mocked_sock)
+    http_response.begin()
+
+    return http_response
 
 
 @pytest.mark.parametrize(

